bigtable-hbase-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -819,93 +836,172 @@ public boolean isTableAvailable(TableName tableName, byte[][] splitKeys) throws
   @Override
   public void snapshot(String snapshotName, TableName tableName)
       throws IOException, SnapshotCreationException, IllegalArgumentException {
-    throw new UnsupportedOperationException("snapshot");  // TODO
+    com.google.longrunning.Operation operation = snapshotTable(snapshotName, tableName);
+    try {
+      waitForOperation(operation.getName());
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new IOException("Create snapshot operation was interrupted", e);
+    }
+  }
+
+  private com.google.longrunning.Operation snapshotTable(String snapshotName, TableName tableName)
+      throws IOException {
+    BigtableClusterName clusterName = this.connection.getSession().getClusterName();
```

Comment:
Why are we getting BigtableClusterName and BigtableInstanceName from two different places? Are they V1 and V2 names? If both are V2 name we can get InstanceName from ClusterName.